fmriprep/workflows/confounds.py

```diff
@@ -243,3 +262,197 @@ def reverse_order(inlist):
     if isinstance(inlist, list):
         inlist.reverse()
     return inlist
+
+
+def get_ica_confounds(ica_out_dir):
+    import os
+    import numpy as np
+    from niworkflows.nipype import logging
+
+    # To catch edge cases when there are either no noise or signal components
+    LOGGER = logging.getLogger('workflow')
+
+
+    # Pass in numpy array and column base name to generate headers
+    # modified from add_header_func
+    def aroma_add_header_func(np_arr, col_base, comp_nums):
+        import pandas as pd
+        from sys import version_info
+        PY3 = version_info[0] > 2
+
+        df = pd.DataFrame(np_arr, columns=[str(col_base) + str(index) for index in comp_nums])
+        df.to_csv(str(col_base) + "ica_confounds.tsv",
+                  sep="\t" if PY3 else '\t'.encode(), index=None)
+
+        return os.path.abspath(str(col_base) + "ica_confounds.tsv")
+
+    # Partial regression (close to, but not identical to fsl_regfilt)
+    def calc_residuals(x, y):
+        X = np.column_stack(x + [[0] * len(x[0])])
```

Comment:
You're right, the intended purpose of this was to concatenate a column of zeros to the coefficient matrix, but this does not do that, and furthermore, I believe I should be [appending a column of ones](https://docs.scipy.org/doc/numpy/reference/generated/numpy.linalg.lstsq.html) to the matrix instead of a column of zeros (sorry, my matrix algebra is lacking). If my thought process is correct, the command should like this:
```
X = np.concatenate((x, np.ones((x.shape[0], 1))), axis=1)
```
Thus X would look something like:
```
[ 2.34, 4.23, 1],
[ 6.23, 1.33, 1],
[ 2.67, 9.21, 1],
...
[ 3.56, 5.34, 1]
```
That fits better with the numpy example provided in the link.